packages/react-i18n/src/utilities/interpolate.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Used to interpolate a string with values from an object.
+ * `{key}` will be replaced with `replacements[key]`.
+ */
+export const DEFAULT_INTERPOLATION = /({([^}]*)})/g;
+
+/**
+ * Similar to Shopify themes' locale files.
+ * `{{key}}` will be replaced with `replacements[key]`.
+ */
+export const MUSTACHE_INTERPOLATION = /({{[\s]*([\S]+?)[\s]*}})/g;
+
+/**
+ * Similar to Ruby ERB templating system.
+ * `<%= key %>` will be replaced with `replacements[key]`.
+ */
+export const ERB_INTERPOLATION = /(<%=[\s]*([\S]+?)[\s]*%>)/g;
```

Comment:
We officially only support the default format, but now offer a way to override it. If anyone decides to override the default format, it's their responsibility to make sure the format regex works well.

Though I still wanted to make sure our implementation wasn't assuming the format anywhere, which wasn't the case before (substring stuff here and there), by testing with different formats.

Since I took the time the write a couple regexes, I believe it's safer to expose them rather then to let devs figure it out every time.

Reply:
As a note for anyone else who might see this and want to use a custom regex, you'll also need to reach out to INTL Languages to help configure your Translation Platform `translation.yml` with a custom placeholder regex.

https://development.shopify.io/engineering/developing_at_Shopify/internationalization/translation_platform#Component_Scheme_Placeholders